lib/routes/charge/pos_invoice.dart

```diff
@@ -763,7 +769,12 @@ class POSInvoiceState extends State<POSInvoice> with TickerProviderStateMixin {
         }).then((res) {
       if (res?.paid == true) {
         Navigator.of(context).push(TransparentPageRoute((context) {
-          return SuccessfulPaymentRoute();
+          return SuccessfulPaymentRoute(
+            currentUser: user,
```

Comment:
All these additional parameters only passed from POS flow complicates the widget and all its children hierarchy.
We should find a. better and simple way to separate the flows such as  pass a widget for the final desired "SuccesfullPaymentMessage" to override the default behaviour.

Reply:
[Something like this?](https://github.com/breez/breezmobile/pull/405/commits/522e4b2b5a838796bc79ca1ce63225468ea0bf1d)

Reply:
I don't see why we have two classes here instead of a method that builds SuccesfullPaymentMessage.

Reply:
@erdemyerebasmaz  this is better than before although I think it is better to only pass a function (print()) instead of print parameters that needs to go all the way down.

Reply:
[Done](https://github.com/breez/breezmobile/pull/405/commits/56ab8d522d3ef67a2f3d91f3adbe8a2c114fe776)